core/services/ocr3/promwrapper/types.go

```diff
@@ -0,0 +1,51 @@
+package promwrapper
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type functionType string
+
+const (
+	query               functionType = "query"
+	observation         functionType = "observation"
+	validateObservation functionType = "validateObservation"
+	outcome             functionType = "outcome"
+	reports             functionType = "reports"
+	shouldAccept        functionType = "shouldAccept"
+	shouldTransmit      functionType = "shouldTransmit"
+)
+
+var (
+	buckets = []float64{
```

Comment:
Are these buckets applicable for all chains? Solana might be significantly faster than Ethereum. I wonder if we should have buckets per chain.

Reply:
For CCIP it shouldn't be a big deal. Most of the work we do in OCR is reading from the database (ChainReader), doing some data transformation, reaching 3rd party APIs etc. There is still some communication with the chain involved but it shouldn't be that relevant. However, we might come back to that after seeing different values on non-evm chains

Reply:
To answer your question: yes, buckets are applicable for all chains